### ISSUE TITLE
Rule Proposal: no-duplicate-exports

### ISSUE THREAD

**Issue by thomassuckow** [eslint/eslint#6573]
The rule will enforce not overriding previous exports within a module

The following patterns are considered problems:

```js
/*eslint no-duplicate-exports: "error"*/

export default function() {}
export default class Foo {}

export { name1, name2, name3 };
export { name4 as name3 };
```

The following patterns are not considered problems:
```js
/*eslint no-duplicate-exports: "error"*/

export default function() {}

export { name1, name2, name3 };
export { name4 as name5 };
```

This prevents the unexpected override of exports (such as from copy pasting of the default export) that result in a different object being imported than expected.

This is a good companion to no-duplicate-imports and should be distributed with it.

Unfortunately I do not have time to create this rule myself at this time.

**Comment by vitorbal** [comment]
I think the `export` rule of the [eslint-plugin-import](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/export.md) does what you are looking for.

A separate question would be if the team considers that this should be part of the core or if it's fine in a plugin. @eslint/eslint-team what do you guys think?

**Comment by btmills** [comment]
I'm reading [15.2.1.1](http://www.ecma-international.org/ecma-262/6.0/#sec-module-semantics-static-semantics-early-errors) as saying this should be a syntax error. Specifically:

> It is a Syntax Error if the ExportedNames of ModuleItemList contains any duplicate entries.

Perhaps related, we have eslint/espree#127, which deals with duplicate `let` and `const` names that should also be a syntax error.

**Comment by michaelficarra** [comment]
That's correct, this is a syntax error. No need for a lint rule.

**Comment by thomassuckow** [comment]
Thanks for finding it in the spec. I'll redirect my efforts to babel
